subsys/bluetooth/services/fast_pair/fp_storage/include/fp_storage_ak_bond.h

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef _FP_STORAGE_AK_BOND_H_
+#define _FP_STORAGE_AK_BOND_H_
+
+#include <sys/types.h>
+#include <stdbool.h>
+#include <zephyr/bluetooth/addr.h>
+
+#include "fp_common.h"
+
+/**
+ * @defgroup fp_storage_ak_bond Fast Pair storage of associations between Account Keys and BLE bonds
+ * @brief Internal API for Fast Pair storage of associations between Account Keys and BLE bonds
+ *
+ * Used only when the CONFIG_BT_FAST_PAIR_STORAGE_AK_BOND Kconfig option is enabled.
+ *
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** Create and save the Account Key association with the potential Bluetooth bond that is being
+ *  established from the active connection.
+ *
+ *  The module links the Account Key information with the Bluetooth bond and stores
+ *  it in the NVM memory. The association and the Bluetooth bond will be deleted if the bond is not
+ *  confirmed using the @ref fp_storage_ak_bond_conn_confirm API before the bonding process is
+ *  finalized using the @ref fp_storage_ak_bond_conn_finalize API.
+ *
+ * @param[in] conn_ctx Connection context associated with the potential the Bluetooth bond.
+ * @param[in] addr The Bluetooth address of the potential bond.
+ * @param[in] account_key Account Key that the bond is associated with. It can be set to NULL in
+ *			  case of the initial pairing Procedure, when the Account Key is not known
+ *			  yet.
+ *
+ * @return 0 If the operation was successful. Otherwise, a (negative) error code is returned.
+ */
+int fp_storage_ak_bond_conn_create(const void *conn_ctx, const bt_addr_le_t *addr,
+				   const struct fp_account_key *account_key);
+
+/** Indicate that the potential Bluetooth bond has been confirmed.
+ *
+ *  It confirms the potential Bluetooth bond associated with the Account Key. The association should
+ *  be earlier created using the @ref fp_storage_ak_bond_conn_create API.
+ *
+ * @param[in] conn_ctx Connection context associated with the Bluetooth bond.
+ * @param[in] addr The Bluetooth address of the bond.
+ */
+void fp_storage_ak_bond_conn_confirm(const void *conn_ctx, const bt_addr_le_t *addr);
+
+/** Indicate that the bonding process has been finalized.
+ *
+ *  It relates to the Account Key association with the Bluetooth bond that was created
+ *  earlier using the @ref fp_storage_ak_bond_conn_create API.
+ *
+ * @param[in] conn_ctx Connection context associated with the Bluetooth bond.
+ */
+void fp_storage_ak_bond_conn_finalize(const void *conn_ctx);
```

Comment:
The name suggest that procedure is finalized on disconnection. Maybe we can call it e.g. `fp_storage_ak_bond_conn_disonnected`/`fp_storage_ak_bond_conn_drop`?